sql/catalyst/src/main/scala/org/apache/spark/sql/types/UserDefinedType.scala

```diff
@@ -74,7 +74,7 @@ abstract class UserDefinedType[UserType] extends DataType with Serializable {
   /**
    * The default size of a value of the UserDefinedType is 4096 bytes.
    */
-  override def defaultSize: Int = 4096
+  override def defaultSize: Int = sqlType.defaultSize
```

Comment:
Why did you change this one? Also doesn't match comment anymore (I think specifying the value in the comment is redundant and not useful)